app/settings.go

```diff
@@ -50,8 +71,74 @@ func (s *settings) apply() {
 	}
 }
 
+func (s *settings) load() {
+	err := s.loadFromFile(s.schema.StoragePath())
+	if err != nil {
+		fyne.LogError("Settings load error:", err)
+	}
+
+	s.setupTheme()
+}
+
+func (s *settings) loadFromFile(path string) error {
+	file, err := os.Open(path)
+	if err != nil {
+		if os.IsNotExist(err) {
+			os.MkdirAll(filepath.Dir(path), 0700)
+			return nil
+		}
+		return err
+	}
+	decode := json.NewDecoder(file)
+
+	return decode.Decode(&s.schema)
+}
+
+func (s *settings) setupTheme() {
+	name := s.schema.ThemeName
+	env := os.Getenv("FYNE_THEME")
```

Comment:
How about:

```go
if envTheme := os.Getenv("FYNE_THEME"); envTheme != nil {
  …
}
```

Reply:
then envTheme is not in scope for the code below surely?

Reply:
Oh sorry, I see it now :)